### ISSUE TITLE
MSVC fails to build profiles.c on 8.8 branch

### ISSUE THREAD

**Issue by angelmixu** [libvips/libvips#1391]
Hi!

We are migrating (or trying to... xD) to the new vips8 C++ interface, and when using the libvips 8.8 branch, I had to add profiles.c file and some other new files to the solution we are using (it's this one: https://github.com/angelmixu/libvips-vs2017/tree/vips8.8).

So, now Visual Studio 2019 fails to compile with this error:

```
1>------ Build started: Project: libvips, Configuration: Debug Win32 ------
1>profiles.c
1>C:\code\sampi\ossSources\vips-git-vs2017\srclibvips\libvips\colour\profiles.c(869,1): error C1091:  compiler limit: string exceeds 65535 bytes in length
1>Done building project "libvips.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```
It's a limitation of Visual Studio compiler:, https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1091?view=vs-2019

After looking around, I've found that a good fix is using an array instead of an string.
In Löve 2D they had to revert exactly this kind of issue: https://bitbucket.org/rude/love/commits/e1566aa361551620de1d0e110cd8630fd407db48

I have taken a look at the implicated code, and the profiles are being loaded as a base64 encoded string, and I don't know if there's another function for loading arrays like that.
Also, I'm not sure how to generate the arrays from the original profile.

If someone could guide me with this, I could send a PR. 
Or if you have any other suggestion, or feel that it's a trivial change... I'll gladly accept the fix :)

EDIT: here would be another solution:
http://en.it1352.com/article/b72d8c558d73428a880df447320ad295.html

**Comment by jcupitt** [comment]
Oh no! How annoying.

The files are generated by this shell-script:

https://github.com/libvips/libvips/blob/master/libvips/colour/wrap-profiles.sh

From this part of the build system:

https://github.com/libvips/libvips/blob/master/libvips/colour/Makefile.am#L41

Love codes up binary strings as:

```C
const unsigned char audio_lua[] = {
	0x66,0x75,0x6E,0x63,0x74,0x69,0x6F,0x6E,0x20,0x6C,0x6F,0x76,0x65,0x2E,0x61,
    	0x66,0x75,0x6E,0x63,0x74,0x69,0x6F,0x6E,0x20,0x6C,0x6F,0x76,0x65,0x2E,0x61,
...
}
```

I suppose we'll have to do something similar. 

I found this:

```
hexdump -v -e '" 0x" 1/1 "%02X,"' some-binary-file | fmt
```

Generates:

```
 0x45, 0x29, 0xE7, 0xBE, 0x4D, 0x21, 0x3D, 0x01, 0x3D, 0x69, 0x41, 0x54,
 0xA4, 0x90, 0xD0, 0xBF, 0xC2, 0x40, 0xE6, 0x9B, 0xB5, 0x7D, 0x4D, 0x19,
 0x0A, 0x77, 0x67, 0xE5, 0xEF, 0xC5, 0x3B, 0xCC, 0x8F, 0xD2, 0x8B, 0x06,
 0x87, 0xFF, 0xD9,
```

I'll make it do that instead.

**Comment by jcupitt** [comment]
Here's a branch to experiment in:

https://github.com/libvips/libvips/tree/msvc-friendly-profiles/libvips

It makes `profiles.c` using literal arrays now (an 80,000 line file!!), but I've not yet adjusted `profile_load.c` to match.

**Comment by jcupitt** [comment]
OK, that branch seems to work now for me. Would you be able to test it?

**Comment by angelmixu** [comment]
Thanks @jcupitt !!!
I'll try it tomorrow :)

**Comment by angelmixu** [comment]
Hi @jcupitt !
It builds perfectly and passes our tests, thanks again! :)

**Comment by jcupitt** [comment]
Great, thanks for testing. I've merged to 8.8, this will be in 8.8.2.
